tests/ttnn/unit_tests/operations/test_clone.py

```diff
@@ -0,0 +1,254 @@
+    dtype: str
+        The string representation of the data type (e.g., 'bfloat16', 'float32', 'int32').
+
+    Returns:
+    Corresponding library-specific dtype or None if not found.
+    """
+    dtype_map = {
+        "bfloat16": lib.bfloat16,
+        "float32": lib.float32,
+        "int32": lib.int32,
+    }
+    return dtype_map.get(dtype, None)
+
+
+def run_clone(
+    shape,
+    input_memory_config,
+    output_memory_config,
+    input_dtype,
+    output_dtype,
+    tilized,
+    device,
+):
+    """
+    Function to test the clone operation on NPU. Generates random input data, clones it on NPU,
+    and compares the output with the CPU clone for correctness.
+
+    Parameters:
+    shape: tuple
+        Shape of the input tensor.
+    input_memory_config: MemoryConfig
+        Memory configuration for the input tensor on NPU.
+    output_memory_config: MemoryConfig
+        Memory configuration for the output tensor on NPU.
+    input_dtype: str
+        Data type of the input tensor ('int32' or other).
+    output_dtype: str or None
+        Data type of the output tensor (must be None or match input_dtype when not tilized).
+    tilized: bool
+        Whether to use TILE_LAYOUT or ROW_MAJOR_LAYOUT for NPU tensor.
+    device: ttnn.device
+        Device where the operation is performed (e.g., NPU device).
+
+    Raises:
+    pytest.skip: When certain conditions on dtype mismatch or layout are not met.
+    """
+    if input_dtype == "int32":
+        cpu_input = torch.randint(low=-10, high=11, size=shape, dtype=get_lib_dtype(torch, input_dtype))
+    else:
+        cpu_input = 2 * torch.rand(size=shape, dtype=get_lib_dtype(torch, input_dtype)) - 1
+
+    if input_dtype == "int32":
+        if output_dtype and output_dtype != "int32":
+            pytest.skip("For int32 input, output_dtype must be None or int32.")
+    if output_dtype == "int32" and input_dtype != "int32":
+        pytest.skip("For int32 output, input_dtype must also be int32.")
+    if output_dtype != input_dtype and output_dtype and not tilized:
+        pytest.skip("When not tilized, dtype conversion is not supported.")
+
+    npu_input = to_npu(
+        cpu_input,
+        device,
+        npu_dtype=get_lib_dtype(ttnn, input_dtype),
+        npu_layout=ttnn.TILE_LAYOUT if tilized else ttnn.ROW_MAJOR_LAYOUT,
+    ).to(device, input_memory_config)
+
+    npu_output = ttnn.clone(
+        npu_input,
+        dtype=get_lib_dtype(ttnn, output_dtype),
+        memory_config=output_memory_config,
+    )
+
+    cpu_output = to_cpu(npu_output, shape)
+
+    passing, out = comp_allclose_and_pcc(torch.ops.aten.clone(cpu_input), cpu_output, rtol=0.01, atol=0.01)
+    logger.info(out)
+    assert passing
+
+
+memory_config_list = [
+    ttnn.MemoryConfig(ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM),
+    ttnn.MemoryConfig(ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.L1),
+]
+
+
+@pytest.mark.parametrize(
```

Comment:
Could we get a shape or two with an odd -1 dim? Otherwise, I'm happy with this PR.

Reply:
@jaykru-tt, Actually, I don't quite understand your point. Could you please provide more details?

Reply:
Oh, it turns out you want to add odd last-dimension test cases. I added them as requested.